kotlin/workflow-core/src/main/java/com/squareup/workflow/WorkflowContext.kt

```diff
@@ -175,6 +179,59 @@ inline fun <StateT : Any, OutputT : Any, reified T> WorkflowContext<StateT, Outp
     handler
 )
 
+/**
+ * Subscribes to [flow] and invokes [handler] when it emits the next value.
+ *
+ * This allows us to use this method to, for example, monitor a stream of values emitted by a single
+ * fow over a series of state transitions. (Use [key] if you need to work simultaneously with
+ * several [Flow]s of the same type.)
+ * The subscription will be disposed when `compose` returns without passing a [Flow] of the same
+ * type and with the same key to this method.
+ *
+ * ## Experimental Status
+ *
+ * The [Flow] API is still marked experimental, and the compiler will warn you if you use it without
+ * opting in. The Channels API is also experimental, however Flow is much newer than channels,
+ * so this method is marked as experimental to discourage use in production code.
+ *
+ * @param key An optional string key that is used to distinguish between subscriptions of the same
+ * type.
+ */
+@FlowPreview
+inline fun <StateT : Any, OutputT : Any, reified T> WorkflowContext<StateT, OutputT>.onNext(
+  flow: Flow<T>,
+  key: String = "",
+  noinline handler: (ChannelUpdate<T>) -> WorkflowAction<StateT, OutputT>
+) = onNext(flow, KTypes.fromGenericType(Flow::class, T::class), key, handler)
+
+/**
+ * Subscribes to [flow] and invokes [handler] when it emits the next value.
+ *
+ * This allows us to use this method to, for example, monitor a stream of values emitted by a single
+ * fow over a series of state transitions. (Use [key] if you need to work simultaneously with
+ * several [Flow]s of the same type.)
+ * The subscription will be disposed when `compose` returns without passing a [Flow] of the same
+ * type and with the same key to this method.
+ *
+ * ## Experimental Status
+ *
+ * The [Flow] API is still marked experimental, and the compiler will warn you if you use it without
```

Comment:
I think I feel okay merging it so long as this warning stays in tact.

Is there any practical way we can move this method out to be an extension from a new `workflow-flow` module though?

Reply:
We can totally do that. Doing so makes sense for safety (you need to opt in to even see the support), but long-term I definitely think it should live in this module, since `Flow` will be as, if not more, fundamental a currency for streams than channels. And I wasn't sure it was worth the hassle of creating separate modules just for safety when we already have the annotations.

How strongly do you feel about making a separate module?

Reply:
I think we should do the module. The tie breaker for me is forcing people to take in a transitive dependency on flow in order to use our core stuff, and the risk of its inevitably incompatible releases screwing up our own release cadence and backward compatibility. 

I'd really like to minimize how much baggage is required to adopt our work.

Reply:
If there were a transitive dependency on some additional library, I would agree with you. Flow is included in the core, common coroutines module though, just like channels.

Reply:
Oh, okay then. That was my only valid argument.